src/Microsoft.SqlTools.SqlCore/ObjectExplorer/SmoModel/SmoTreeNodes.cs

```diff
@@ -5102,5 +4457,411 @@ public override TreeNode CreateChild(TreeNode parent, object context)
         }
     }
 
+    internal static class NodeFilterPropertyHelper
+    {
+        internal static NodeFilterProperty[] GetFilterProperties(object context, NodeTypes? nodeType)
+        {
+            var filterProperties = new List<NodeFilterProperty>();
+            switch (nodeType)
+            {
+					case NodeTypes.Databases:
+							filterProperties.Add(new NodeFilterProperty
+							{
+								Name = "Name",
+								DisplayName = SR.FilterName,
```

Comment:
Can we define a constant and add it to filterProperties since these values are repeated for each nodeType? This would help reduce code duplication.